api/config/v1alpha1/shared_types.go

```diff
@@ -40,23 +40,53 @@ const (
 
 // KubernetesDeploymentSpec defines the desired state of the Kubernetes deployment resource.
 type KubernetesDeploymentSpec struct {
+	// Pod defines the desired replicas, annotations and securityContext of container.
+	//
+	// +optional
+	Pod *KubernetesPodSpec `json:"pod,omitempty"`
+
+	// Container defines the resources and securityContext of container.
+	//
+	// +optional
+	Container *KubernetesContainerSpec `json:"container,omitempty"`
+
+	// TODO: Expose config as use cases are better understood, e.g. labels.
+}
+
+// KubernetesPodSpec defines the desired state of the Kubernetes pod resource.
+type KubernetesPodSpec struct {
 	// Replicas is the number of desired pods. Defaults to 1.
 	//
 	// +optional
 	Replicas *int32 `json:"replicas,omitempty"`
```

Comment:
Replicas should be part of `KubernetesDeploymentSpec`

Reply:
did it